nservicebus/tools/migrate-to-native-delivery.md

```diff
@@ -0,0 +1,161 @@
+                        --dialect <sqlDialect>
+                        [--endpoint] <endpointName>
+                        [--allendpoints]
+```
+
+#### Running a migration
+
+Migrating from RavenDB persistence
+
+```
+dotnet migrate-timeouts ravendb
+                        -t|--target <targetConnectionString>
+                        -c|--cutofftime <cutofftime>
+                        --serverUrl <serverUrl>
+                        --databaseName <databaseName>
+                        [--prefix] <prefix>
+                        [--ravenVersion] <ravenVersion>
+                        [--endpoint] <endpointName>
+                        [--allendpoints]
+                        [-a|--abort]
+
+```
+
+Migrating from Sql persistence
+
+```
+dotnet migrate-timeouts preview sqlp
+                        -t|--target <targetConnectionString>
+                        -c|--cutofftime <cutofftime>
+                        --source <source>
+                        --tablename <tablename>
+                        --dialect <sqlDialect>
+                        [--endpoint] <endpointName>
+                        [--allendpoints]
+                        [-a|--abort]
+
+```
+
+Aborting a migration
+
+```
+dotnet migrate-timeouts <repeat previous parameters>
+                        [-a|--abort]
+
+```
+
+## How the tool works
+
+Before actually migrating any timeouts, it's highly recommended to use the preview option.
+This option will perform the following actions:
+ - check that the tool can connect to the storage
+ - check that the tool can connect to the target transport
+ - verify that the necessary topology is in place in the target transport to support native delayed delivery
+ - list all the endpoints for which the tool can detect timeouts
+ - calculate the amount of timeouts to migrate per endpoint
+ - validate if there are timeouts the tool is [unable to migrate](migrate-to-native-delivery.md#limitations)
+
+Once this information has been reviewed, the migration process can be initialized.
+Even though the tool supports migrating all endpoints connected to the persistence at once, it is highly suggested to migrate endpoint by endpoint, especially for critical endpoints. Even when selecting the -allendpoints options, the tool will conduct an endpoint-by-endpoint migration behind the scenes.
+
+## Limitations
+
+As documented in the [RabbitMQ transport](/transports/rabbitmq/delayed-delivery.md), the maximum delay value of a timeout is 8,5 years. If the migration tool encounters any timeouts that have delivery times set beyond that, it won't be possible to migrate that endpoint.
+
+If the tool presents endpoints that are not part of the system when running the --analyze option, it might be that an endpoint was renamed at some point.
+Any timeouts that were stored for that endpoint, might already be late in delivery and should be handled seperate from the migration tool since the tool has no way to detect where to migrate them to.
```

Comment:
☝️ this doesn't make sense to me

Reply:
Well let's say that the customer has EndpointA, they renamed the endpoint to "Shipping", but never checked that there were still timeouts waiting to be sent out by the timeout manager. Those timeouts have OwningTimeoutManager set to EndpointA, not to Shipping. In that case we will find timeouts for EndpointA, but EndpointA is not part of the system anymore. It's not the responsibility of the tool to fix these, and the customer might not even realise that these are a problem. Does that make more sense?